src/MonitorProfiler/Environment/EnvironmentHelper.cpp

```diff
@@ -3,12 +3,30 @@
 // See the LICENSE file in the project root for more information.
 
 #include "EnvironmentHelper.h"
+#include "../Logging/LogLevelHelper.h"
 #include "productversion.h"
 
 using namespace std;
 
 #define IfFailLogRet(EXPR) IfFailLogRet_(pLogger, EXPR)
 
+HRESULT EnvironmentHelper::GetDebugLoggerLevel(
+    const std::shared_ptr<IEnvironment>& pEnvironment,
+    LogLevel& level)
+{
+    HRESULT hr = S_OK;
```

Comment:
I'm assuming I'm missing something since I see this in several places - why do we seemingly set `hr = S_OK` but then never use it?

Reply:
The `IfNullRet`, `IfFailRet`, `IfFailLogRet`, etc macros assume there is a variable named `hr` to which they can set the result of their condition. It makes it so that they don't have to define scoped variables every time and you can inspect the result after the macro as passed.

Reply:
The `IfNullRet`, `IfFailRet`, `IfFailLogRet`, etc macros assume there is a variable named `hr` to which they can set the result of their condition. It makes it so that they don't have to define scoped variables every time and you can inspect the result after the macro as passed.